src/hotspot/share/utilities/ostream.cpp

```diff
@@ -893,17 +890,17 @@ void defaultStream::write(const char* s, size_t len) {
   intx holder = hold(thread_id);
 
   if (DisplayVMOutput &&
-      (_outer_xmlStream == NULL || !_outer_xmlStream->inside_attrs())) {
+      (_outer_xmlStream == nullptr || !_outer_xmlStream->inside_attrs())) {
     // print to output stream. It can be redirected by a vfprintf hook
     jio_print(s, len);
   }
 
   // print to log file
-  if (has_log_file()) {
-    int nl0 = _newlines;
-    xmlTextStream::write(s, len);
```

Comment:
xmlTextStream::write seems to be unused after this change.

Reply:
It does seem that way. The `outputStream::write` method is the function which all subclasses of `outputStream` implements, so we shouldn't remove this function without removing `xmlTextStream`. That's probably a separate RFE, and perhaps not desirable as it'd require merging the class into  `defaultStream`.

Reply:
I'm confused by this part of the change. How can the original code call `xmlTextStream::write` without an instance of `xmlTextStream` to apply it to???

Reply:
`defaultStream` inherits from `xmlTextStream` and `defaultStream::write` exists, so this specifies to call the superclass method, and not its own method. In long form: `this->xmlTextStream::write` (haven't attempted to compile)